### PR TITLE
Pt 154281622 encode calldata

### PR DESCRIPTION
**Pull request by happi**
Provide a function for turning a function call into calldata.

[finishes #154281622]

### CODE REVIEW

File: apps/aering/test/aer_abi_tests.erl [hunk 1]
```diff
@@ -0,0 +1,12 @@
+-module(aer_abi_tests).
+
+-include_lib("eunit/include/eunit.hrl").
+
+encode_call_with_integer_test() -> 
+    <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
```
Comment (lucafavatella) [inline apps/aering/test/aer_abi_tests.erl]: I understand spec is https://github.com/aeternity/protocol/pull/9/files#diff-a7778cdeae90e4a2ccdc834c003760e7R227

Comment (happi) [reply apps/aering/test/aer_abi_tests.erl]: The (offline) spec has been updated as the language was implemented. I have to fix the protocoll repo.

File: apps/aering/src/aer_abi.erl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    %% TODO: check that function exists in cotract.
+    FunctionHandle = encode_function(Contract, Function),
+    Call = {FunctionHandle, Argument},
+    {0, Data} = aer_data:to_binary(Call),
+    _ArgumentType = get_type(Argument),
+    %% TODO: Verify that the type matches the function signature.
+    Data.
+
+encode_function(_Contract, Function) ->
+     << <<X>> || X <- Function>>.
```
Comment (lucafavatella) [inline apps/aering/src/aer_abi.erl]: I understand this is equivalent to `list_to_binary(Function)`?

Comment (happi) [reply apps/aering/src/aer_abi.erl]: Almost. It will truncate values > 255 to 8 bits without crashing.

File: test/ring_bytecode_aevm_SUITE.erl [hunk 3]
```diff
@@ -28,15 +32,15 @@ execute_identy_fun_from_ring_file(_Cfg) ->
     ok = aeb_disassemble:pp(Code),
 
     %% Create the call data
-    Call = {<<"main">>, 42},
-    {0, Data} = aer_data:to_binary(Call),
-    Res =
+    CallData = aer_abi:create_calldata(Code, "main", 42),
+    io:format("CallData: ~p~n", [CallData]),
```
Comment (sennui) [inline test/ring_bytecode_aevm_SUITE.erl]: do you want to leave it over here?

Comment (happi) [reply test/ring_bytecode_aevm_SUITE.erl]: Ill remove in my next branch/PR when it actually will be used.

**Pull request merged**